lib/wickedbase.pm

```diff
@@ -984,6 +985,12 @@ sub post_run {
     $self->check_logs() unless $self->{skip_check_logs_on_post_run};
     $self->check_coredump();
     $self->upload_wicked_logs('post');
+
+    my $time_since_barrier_wait = time - ($self->{fix_barrier_wait} // 0);
+    if ($time_since_barrier_wait < lockapi::POLL_INTERVAL) {
+        bmwqemu::diag("If the other parallel JOB might wait in barrier_wait(), we should not quit this job to early!");
```

Comment:
```suggestion
        bmwqemu::diag("Waiting for parallel job to finish, so barrier will be properly released");
```

Reply:
I need to explain this `fix_barrier_wait`, because what you wrote sounds more like the opposite of what I tried to achieve.

This is the situation. 
```
┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│  GOOD CASE                                                                                       │
│                                                                                                  │
│           Child                                            Parent                                │
│              │                                                 │                                 │
│              │                                                 │                                 │
│              ├────────────►barrier_wait(a)                     │                                 │
│              │                      │                          │                                 │
│              │                      │                          │                                 │
│              │                      │                          │                                 │
│              │                      │                          │                                 │
│              │          api::check_barrier(a)                  │                                 │
│              │            -> not released (409)                ├───────────────────┐             │
│              │                      │                          │                   ▼             │
│              │                      │                          │         Parent::barrier_wait(a) │
│              │                sleep 5                          │                   ▼             │
│              │                      │                          │         api::check_barrier(a)   │
│              │                      │                          │           -> successful   (200) │
│              │                      │                          │                   │             │
│              │                      │                          │◄──────────────────┘             │
│              │                      │                          ▼                                 │
│              │                      │                    do_some_more_tests()                    │
│              │                      │                          │                                 │
│              │          api::check_barrier(a)                  │                                 │
│              │            -> Successful (200)                  │                                 │
│              │                      │                          │                                 │
│              │◄─────────────────────┘                          │                                 │
│              ▼                                                 ▼                                 │
│        Job finished successful                           Job Finished                            │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
                                                      fix_barrier_wait




┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│   BAD CASE                                                                                       │
│                                                                                                  │
│           Child                                            Parent                                │
│              │                                                 │                                 │
│              │                                                 │                                 │
│              ├────────────►barrier_wait(a)                     │                                 │
│              │                      │                          │                                 │
│              │                      │                          │                                 │
│              │           api::check_barrier(a)                 ├───────────────────┐             │
│              │             -> not released (409)               │                   │             │
│              │                      │                          │                   ▼             │
│              │                      │                          │            fix_barrier_wait     │
│              │                sleep 5                          │         Parent::barrier_wait(a) │
│              │                      │                          │                   ▼             │
│              │                      │                          │         api::check_barrier(a)   │
│              │                      │                          │           -> successful   (200) │
│              │                      │                          │                   │             │
│              │                      │                          │◄──────────────────┘             │
│              │                      │                          │                                 │
│              │                      │                          ▼                                 │
│              │          api::check_barrier(a)            Job Finished                            │
│              │            -> Lock-Owner-Finished (410)                                           │
│              │               -> die()                                                            │
│              │                     │                                                             │
│              │                     │                                                             │
│              │◄────────────────────┘                                                             │
│              ▼                                                                                   │
│        Job failed unexpected                                                                     │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
```
## Explanation of `BAD CASE`:
The client job call `barrier_wait()`. In barrier wait is a loop which is calling the `api::check_barrier` call (url is `barrier/$name`). The return value could be

* 200 => barrier released
* 409 => you need to wait - not all expected jobs are here
* 410 => something unexpected happen - cancel with error

The first call of `api::check_barrier()` return 409, because the Parent didn't called barrier_wait() till now.
Then Parent is calling `barrier_wait()` and get's 200. This was the last call in the Parent job, so the Parent job immediately finished.
By default Child process poll the state of barrier every 5 seconds. With this, the next call of `api::check_barrier()` get's 409, because the parent process already finished. which lead to an unexpected die().

=> solution, simply finish not 5seconds after the last barrier_wait() call.

Reply:
I raised this issue also here: https://github.com/os-autoinst/os-autoinst/issues/2340